specs/agents/tracing-spans-service-target.md

```diff
@@ -0,0 +1,244 @@
+## Span service target
+
+The span service target fields replace the `span.destination.service.*` fields that are deprecated.
+
+However, it does not replace [destination fields](https://www.elastic.co/guide/en/ecs/current/ecs-destination.html)
+which are used for APM/SIEM integration and focus on network-level attributes.
+
+- [span.context.destination.address](https://www.elastic.co/guide/en/ecs/current/ecs-destination.html#field-destination-address)
+- [span.context.destination.port](https://www.elastic.co/guide/en/ecs/current/ecs-destination.html#field-destination-port)
+
+APM Agents should now use fields described below in `span.context.service.target`, those are a subset
+of [ECS Service fields](https://www.elastic.co/guide/en/ecs/current/ecs-service.html).
+
+- `span.context.service.target.type` : [ECS service.type](https://www.elastic.co/guide/en/ecs/current/ecs-service.html#field-service-type)
+  , optional.
+- `span.context.service.target.name` : [ECS service name](https://www.elastic.co/guide/en/ecs/current/ecs-service.html#field-service-name)
+  , optional, ignored if `span.context.service.target.type` is not provided.
```

Comment:
From the APM server's perspective, does this mean that for validating `span.context.service.target` it should expect `span.context.service.target.type` to be a required field?

Reply:
Just to be sure, here do you mean 1) in the data sent by agents, 2) to validate data stored to ES or 3) both ? I'll assume 1) here in my answer.

The whole `span.context.service.target` is optional, but I did not thought that we could maybe enforce a mandatory property within it, so maybe having `span.context.service.target.type` mandatory (but potentially empty) could be a good option, would that make things simpler on the server side for ingestion ?

However, one good reason to keep most of them optional is that the mandatory fields tends to stay forever for backwards compatibility after they are added, so while we don't have currently any use-case for that yet, it could help us later.
Also, another reason to stay cautious with mandatory fields is that if there is any extra complexity on the apm-server side, we can always fix it through code but the protocol can't be modified as easily. Another example of similar strategy in another context is with protobuf in version >=3 where all the attributes are optional for similar reasons.

Maybe @felixbarny could enlighten us with the agent protocol history and what should be the best strategy here.

Reply:
After discussing with @SylvainJuge we have a decision to have APM-server expect `span.context.service.target` with at-least one field, `name` or `type`, as non-empty.